apps/src/music/blockly/MusicBlocklyWorkspace.js

```diff
@@ -324,28 +324,54 @@ export default class MusicBlocklyWorkspace {
   }
 
   updateHighlightedBlocks(playingBlockIds) {
+    if (!this.workspace) {
+      this.metricsReporter.logWarning('workspace not initialized.');
+      return;
+    }
     // Clear all highlights.
-    Blockly.mainBlockSpace.getAllBlocks().forEach(block => {
-      Blockly.mainBlockSpace.highlightBlock(block.id, false);
+    this.workspace.getAllBlocks().forEach(block => {
+      this.workspace.highlightBlock(block.id, false);
     });
     // Highlight playing blocks.
     playingBlockIds.forEach(blockId => {
-      Blockly.mainBlockSpace.highlightBlock(blockId, true);
+      this.workspace.highlightBlock(blockId, true);
     });
   }
 
   // Given a block ID, selects that block.
   // Given undefined, unselects all blocks.
   selectBlock(blockId) {
+    if (!this.workspace) {
+      this.metricsReporter.logWarning('workspace not initialized.');
+      return;
+    }
     if (blockId) {
-      Blockly.mainBlockSpace.getBlockById(blockId).select();
+      this.workspace.getBlockById(blockId).select();
     } else {
-      Blockly.mainBlockSpace.getAllBlocks().forEach(block => {
+      this.workspace.getAllBlocks().forEach(block => {
         block.unselect();
       });
     }
   }
 
+  getSelectedTriggerId(blockId) {
+    if (!this.workspace) {
+      this.metricsReporter.logWarning('workspace not initialized.');
+      return undefined;
+    }
+    const block = this.workspace.getBlockById(blockId);
+    if (!block) {
+      return undefined;
+    }
+    const isSelectedBlockTriggerAt =
+      block.type === BlockTypes.TRIGGERED_AT_SIMPLE2;
```

Comment:
> @bencodeorg: For my understanding, do these 2 suffixed block types (I see a couple of them) represent?

Oh yeah, this is a relic of when we were supporting a few different execution models for the sake of comparison and testing ("Simple", "Simple2", "Tracks", etc) - the one we landed on and have used since is "Simple2" so a lot of the blocks are suffixed with "SIMPLE2".

We had plans to continue supporting the other models for a while in case we wanted to switch back or experiment further but I think at this point we're pretty locked into "Simple2" and could probably go in and clean up all of these names (@breville what do you think?)

Reply:
I would be comfortable cleaning up these names, now that we really have settled on `simple2`!